Firebase.CloudFirestore/source/Firebase.CloudFirestore/ApiDefinition.cs

```diff
@@ -0,0 +1,689 @@
+		[Export ("longitude")]
+		double Longitude { get; }
+	}
+
+	interface IListenerRegistration { }
+
+	// @protocol FIRListenerRegistration <NSObject>
+	[Protocol (Name = "FIRListenerRegistration")]
+	interface ListenerRegistration
+	{
+		// @required -(void)remove;
+		[Abstract]
+		[Export ("remove")]
+		void Remove ();
+	}
+
+	// @interface FIRQueryListenOptions : NSObject
+	[BaseType (typeof (NSObject), Name = "FIRQueryListenOptions")]
+	interface QueryListenOptions
+	{
+		// +(instancetype _Nonnull)options;
+		[Static]
+		[Export ("options")]
+		QueryListenOptions Create ();
+
+		// @property (readonly, assign, nonatomic) BOOL includeQueryMetadataChanges;
+		[Export ("includeQueryMetadataChanges")]
+		bool IncludeQueryMetadataChanges { get; }
+
+		// -(instancetype _Nonnull)includeQueryMetadataChanges:(BOOL)includeQueryMetadataChanges;
+		[Export ("includeQueryMetadataChanges:")]
+		QueryListenOptions SetIncludeQueryMetadataChanges (bool includeQueryMetadataChanges);
+
+		// @property (readonly, assign, nonatomic) BOOL includeDocumentMetadataChanges;
+		[Export ("includeDocumentMetadataChanges")]
+		bool IncludeDocumentMetadataChanges { get; }
+
+		// -(instancetype _Nonnull)includeDocumentMetadataChanges:(BOOL)includeDocumentMetadataChanges;
+		[Export ("includeDocumentMetadataChanges:")]
+		QueryListenOptions SetIncludeDocumentMetadataChanges (bool includeDocumentMetadataChanges);
+	}
+
+	// typedef void (^FIRQuerySnapshotBlock)(FIRQuerySnapshot * _Nullable, NSError * _Nullable);
+	delegate void QuerySnapshotHandler ([NullAllowed] QuerySnapshot snapshot, [NullAllowed] NSError error);
+
+	// @interface FIRQuery : NSObject
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject), Name = "FIRQuery")]
+	interface Query
+	{
+		// @property (readonly, nonatomic, strong) FIRFirestore * _Nonnull firestore;
+		[Export ("firestore", ArgumentSemantic.Strong)]
+		Firestore Firestore { get; }
+
+		// -(void)getDocumentsWithCompletion:(FIRQuerySnapshotBlock _Nonnull)completion;
+		[Async]
+		[Export ("getDocumentsWithCompletion:")]
+		void GetDocuments (QuerySnapshotHandler completion);
+
+		// -(id<FIRListenerRegistration> _Nonnull)addSnapshotListener:(FIRQuerySnapshotBlock _Nonnull)listener;
+		[Export ("addSnapshotListener:")]
+		IListenerRegistration AddSnapshotListener (QuerySnapshotHandler listener);
+
+		// -(id<FIRListenerRegistration> _Nonnull)addSnapshotListenerWithOptions:(FIRQueryListenOptions * _Nullable)options listener:(FIRQuerySnapshotBlock _Nonnull)listener;
+		[Export ("addSnapshotListenerWithOptions:listener:")]
+		IListenerRegistration AddSnapshotListener ([NullAllowed] QueryListenOptions options, QuerySnapshotHandler listener);
+
+		// -(FIRQuery * _Nonnull)queryWhereField:(NSString * _Nonnull)field isEqualTo:(id _Nonnull)value;
+		[Export ("queryWhereField:isEqualTo:")]
+		Query WhereEqualsTo (string field, NSObject value);
+
+		[Wrap ("WhereEqualsTo (field, FromObject (value))", true)]
```

Comment:
Any reason why you are making these Wrappers `virtual`?

Reply:
We had this talk before. I forgot to remove the `true` part. Changing it.